src/editor/FindMatches.java

```diff
@@ -0,0 +1,53 @@
+package editor;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class FindMatches extends Thread{
+    private void findMatches() {
+        TextSearch.matchNum = 0;
+        TextSearch.startIndexes.clear();
+        TextSearch.endIndexes.clear();
+
+        String  foundText = TextEditor.textField.getText();
+        String text = TextEditor.textArea.getText();
+
+        while (TextSearch.threadAtWork) {}
```

Comment:
Why is this loop with an empty body? I think, actually, Java optimizer may remove it from bytecode. You may choose a .class file, open in IntelliJ with decompilation and check it.